src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -492,28 +494,42 @@ export async function makeUtxoEngine(
     },
 
     async sweepPrivateKeys(spendInfo: EdgeSpendInfo): Promise<EdgeTransaction> {
-      const privateKeys = spendInfo.privateKeys ?? []
-      const dummyNumbWalletInfo: NumbWalletInfo = {
+      // Create the promise to return
+      let success: (
+        value: EdgeTransaction | PromiseLike<EdgeTransaction>
+      ) => void
+      let failure: (err?: any) => void
+      const end = new Promise<EdgeTransaction>((resolve, reject) => {
+        success = resolve
+        failure = reject
+      })
+
+      // The temporary wallet info should include all formats for the currency
+      const allFormats = engineInfo.formats
+      if (allFormats == null) {
```

Comment:
Why do you need to check for null? I thought "engineInfo.formats" should be strictly typed to exists no?

Reply:
No. The current type is optional.